library/cc/stream_callbacks.cc

```diff
@@ -0,0 +1,85 @@
+#include "stream_callbacks.h"
+
+// NOLINT(namespace-envoy)
+
+#include "bridge_utility.h"
+
+EnvoyHttpCallbacksAdapter::EnvoyHttpCallbacksAdapter(ExecutorSharedPtr executor,
+                                                     StreamCallbacksSharedPtr callbacks)
+    : executor_(executor), stream_callbacks_(callbacks) {}
+
+envoy_http_callbacks EnvoyHttpCallbacksAdapter::as_envoy_http_callbacks() {
+  envoy_http_callbacks callbacks{
+      .on_headers = &EnvoyHttpCallbacksAdapter::dispatch_on_headers,
+      .on_data = &EnvoyHttpCallbacksAdapter::dispatch_on_data,
+      // on_metadata is not used
```

Comment:
is this intentional? i notice that `on_metadata` isn't used in the kotlin callbacks

Reply:
originally proposed in spdy, disallowed in HTTP/2, proposal for HTTP/2. going to keep it piped up to here but nothing more